### ISSUE TITLE
Why is the loss and accuracy only updated every 100 steps in the Bayesian neural network example?

### ISSUE THREAD

**Issue by nbro** [tensorflow/probability#634]
In the Bayesian neural network example https://github.com/tensorflow/probability/blob/master/tensorflow_probability/examples/bayesian_neural_network.py, why do you run `elbo_loss` and `accuracy` only every 100 steps? Why not at every step?

```
if step % 100 == 0:
    loss_value, accuracy_value = sess.run([elbo_loss, accuracy], feed_dict={handle: train_handle})
```

See line https://github.com/tensorflow/probability/blob/master/tensorflow_probability/examples/bayesian_neural_network.py#L306.

**Comment by Pyrsos** [comment]
As I understand this, it is just for monitoring purposes. You do not necessarily want to print out these values at every step because you will just end up flooding the terminal.

**Comment by nbro** [comment]
@Pyrsos If the loss and accuracy are evaluated at every 100 steps, aren't we effectively reducing the number of training steps?

**Comment by Pyrsos** [comment]
@nbro The train operation happens in the lines above and it does occur at each step: https://github.com/tensorflow/probability/blob/716e76c639d47b3626bcaaa6af91fa3498887595/tensorflow_probability/examples/bayesian_neural_network.py#L302

The only thing that line 306 does is return the `elbo_loss` and `accuracy` values as floats so they can be printed out. It does not update the model's parameters.

**Comment by nbro** [comment]
@Pyrsos Well, I thought we had to evaluate the accuracy but **especially** the loss at every step of the computation. However, I realise now that `train_op = optimizer.minimize(elbo_loss)`, so when we evaluate the training operation, we are essentially minimizing the loss.
